pkgs/by-name/wo/wordlists/package.nix

```diff
@@ -1,45 +1,61 @@
 { lib
-, callPackage
+, symlinkJoin
 , nmap
 , rockyou
-, runtimeShell
 , seclists
-, symlinkJoin
-, tree
 , wfuzz
 , lists ? [
     nmap
     rockyou
     seclists
     wfuzz
   ]
+, writeShellScriptBin
+, tree
 }:
+let
+  version = "unstable-2023-11-14";
 
-symlinkJoin rec {
-  pname = "wordlists";
-  version = "unstable-2023-10-10";
-
-  name = "${pname}-${version}";
-  paths = lists;
+  wordlistsCollection = symlinkJoin {
+    name = "wordlists-collection-${version}";
+    paths = lists;
 
-  postBuild = ''
-    mkdir -p $out/bin
+    postBuild = let
+      pathsToKeep = lib.pipe [
+        "$out/share/wordlists/*"
+        "$out/share/wordlists"
+        "$out/share"
+        "$out"
+      ] [
+        (map (p: ''-path "${p}"''))
+        (builtins.concatStringsSep " -o ")
+      ];
+    in ''
+      find $out -type d ! \( ${pathsToKeep} \) -printf "rm -r \"%p\" 2>/dev/null || true\n" | sh
+    '';
```

Comment:
isn't the only path we want to keep `"$out/share/wordlists/*"`?
and the find scirpt part is a mildly terrifying 🙈

Reply:
Yeah, I'm not a fan of it either.

I played around with it for a long time. I found that if I keep the current arguments but only match `"$out/share/wordlists/*"`, it will go ahead and delete `$out`, `$out/share`, and `$out/share/wordlists`. And it will also do so in the order specified, meaning the two last commands end up failing, which is why the `|| true` is there. Now `-exec` isn't available anymore, because `|| true` would be parsed as part of `find` ... and so on.

If you're able to come up with anything shorter, I'll happily replace this!

Reply:
@pbsds Can I ask you to have a look at this? Maybe you know how to do this in a cleaner way?